### ISSUE TITLE
有关代码的一些问题

### ISSUE THREAD

**Issue by Dimitri666** [weicy15/CGI#5]
您好，关于code，有一些问题：
`for i in range(self.L):
            cur_embedding = torch.mm(self.joint_adjaceny_matrix_normal, cur_embedding)
            all_embeddings.append(cur_embedding)

            # edge_num * 2emebdding_size
            edge_cat_embedding = torch.cat([cur_embedding[self.row], cur_embedding[self.col]], dim=-1)
            # edge_num
            edge_mask = self.edge_mask_learner[i](edge_cat_embedding)
            bias = 0.0 + 0.0001  # If bias is 0, we run into problems
            eps = (bias - (1 - bias)) * torch.rand(edge_mask.size()) + (1 - bias)
            edge_gate_inputs = torch.log(eps) - torch.log(1 - eps)
            edge_gate_inputs = edge_gate_inputs.to(self.device)
            edge_gate_inputs = (edge_gate_inputs + edge_mask) / self.choosing_tmp
            edge_mask = torch.sigmoid(edge_gate_inputs).squeeze(1)
            edge_mask_list.append(edge_mask)

            # user_num + item_num
            node_mask = self.node_mask_learner[i](cur_embedding)
            bias = 0.0 + 0.0001  # If bias is 0, we run into problems
            eps = (bias - (1 - bias)) * torch.rand(node_mask.size()) + (1 - bias)
            node_gate_inputs = torch.log(eps) - torch.log(1 - eps)
            node_gate_inputs = node_gate_inputs.to(self.device)
            node_gate_inputs = (node_gate_inputs + node_mask) / self.choosing_tmp
            node_mask = torch.sigmoid(node_gate_inputs)
            node_mask_list.append(node_mask)

        all_embeddings = torch.stack(all_embeddings, dim=0)
        all_embeddings = torch.mean(all_embeddings, dim=0, keepdim=False)
        user_embeddings, item_embeddings = torch.split(all_embeddings, [self.user_num,self.item_num])`

1. GCN中，每一层使用输入的cur_embedding都是一样的，感觉应该是后一层的输入emb是前一层的输出emb，也就是mask之后的emb。而且每层的mask都是通过同一个emb得到的。不确定这个会不会有影响？

line 285
`        for i in range(self.L):
            edge_mask = edge_mask_list[i]
            new_edge = torch.mul(self.joint_adjaceny_matrix_normal.values(), edge_mask)

            cur_embedding_edge_drop = torch_sparse.spmm(self.joint_adjaceny_matrix_normal.indices(), new_edge, self.user_num + self.item_num, self.user_num + self.item_num, cur_embedding_edge_drop)
            all_embeddings_edge_drop.append(cur_embedding_edge_drop)`
2. 推理过程直接将mask和邻接矩阵相乘，这里并没有像论文说的，去掉概率小于0.5的Node和edge。不确定这个有什么问题。

**Comment by weicy15** [comment]
for i in range(self.L):
  cur_embedding = torch.mm(self.joint_adjaceny_matrix_normal, cur_embedding)
  all_embeddings.append(cur_embedding)
在这当中每一层的curembedding已经得到更新

**Comment by weicy15** [comment]
对于第二点来说，增加0.5的截断和原先的方式相比，影响不大

**Comment by Dimitri666** [comment]
> 对于第二点来说，增加0.5的截断和原先的方式相比，影响不大

谢谢答复
